astropy/io/fits/tests/test_connect.py

```diff
@@ -57,13 +60,35 @@ def equal_data(a, b):
     return all(np.all(a[name] == b[name]) for name in a.dtype.names)
 
 
+@contextlib.contextmanager
+def working_directory(path):
+    """
+    A context manager changing the working directory to the given
+    path. The directory is changed back on exit.
+    """
+
+    cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
+
+
 class TestSingleTable:
     def setup_class(self):
         self.data = np.array(
             list(zip([1, 2, 3, 4], ["a", "b", "c", "d"], [2.3, 4.5, 6.7, 8.9])),
             dtype=[("a", int), ("b", "U1"), ("c", float)],
         )
 
+    def test_path(self, tmp_path):
+        with working_directory(tmp_path):
+            filename = "temp.fits"
+            t1 = Table(self.data)
+            t1.write(filename, overwrite=True)
+            t1.write(Path(filename), format="fits", overwrite=True)
```

Comment:
I think there may be another side effect when `format` is not specified, then the path is also converted to a string (saw that yesterday but did not check carefully).

So I think we should have this:
```
            t1.write(filename, format="fits")
            t1.write(filename, format="fits", overwrite=True)
```
first time `overwrite` is not needed, 2nd time it should fail without the fix.

Reply:
So given the other issue with string conversion, I think we should test both writing to a local file (with the context manager) and writing to a path with a directory, i.e. `tmp_path / filename`.